homeassistant/components/motioneye/media_source.py

```diff
@@ -0,0 +1,349 @@
+            can_play=False,
+            can_expand=True,
+            children=[
+                self._build_media_config(entry)
+                for entry in self.hass.config_entries.async_entries(DOMAIN)
+            ],
+            children_media_class=MEDIA_CLASS_DIRECTORY,
+        )
+
+    @classmethod
+    def _build_media_device(
+        cls,
+        config: ConfigEntry,
+        device: dr.DeviceEntry,
+        full_title: bool = True,
+    ) -> BrowseMediaSource:
+        return BrowseMediaSource(
+            domain=DOMAIN,
+            identifier=f"{config.entry_id}#{device.id}",
+            media_class=MEDIA_CLASS_DIRECTORY,
+            media_content_type="",
+            title=f"{config.title} {device.name}" if full_title else device.name,
+            can_play=False,
+            can_expand=True,
+            children_media_class=MEDIA_CLASS_DIRECTORY,
+        )
+
+    def _build_media_devices(self, config: ConfigEntry) -> BrowseMediaSource:
+        """Build the media sources for device entries."""
+        device_registry = dr.async_get(self.hass)
+        devices = dr.async_entries_for_config_entry(device_registry, config.entry_id)
+
+        base = self._build_media_config(config)
+        base.children = [
+            self._build_media_device(config, device, full_title=False)
+            for device in devices
+        ]
+        return base
+
+    @classmethod
+    def _build_media_kind(
+        cls,
+        config: ConfigEntry,
+        device: dr.DeviceEntry,
+        kind: str,
+        full_title: bool = True,
+    ) -> BrowseMediaSource:
+        return BrowseMediaSource(
+            domain=DOMAIN,
+            identifier=f"{config.entry_id}#{device.id}#{kind}",
+            media_class=MEDIA_CLASS_DIRECTORY,
+            media_content_type=MEDIA_CLASS_DIRECTORY,
+            title=(
+                f"{config.title} {device.name} {kind.title()}"
+                if full_title
+                else kind.title()
+            ),
+            can_play=False,
+            can_expand=True,
+            children_media_class=(
+                MEDIA_CLASS_MOVIE if kind == "movies" else MEDIA_CLASS_IMAGE
+            ),
+        )
+
+    def _build_media_kinds(
+        self, config: ConfigEntry, device: dr.DeviceEntry
+    ) -> BrowseMediaSource:
+        base = self._build_media_device(config, device)
+        base.children = [
+            self._build_media_kind(config, device, kind, full_title=False)
+            for kind in MEDIA_CLASS_MAP
+        ]
+        return base
+
+    async def _build_media_path(
+        self,
+        config: ConfigEntry,
+        device: dr.DeviceEntry,
+        kind: str,
+        path: str,
+    ) -> BrowseMediaSource:
+        """Build the media sources for media kinds."""
+        base = self._build_media_kind(config, device, kind)
+
+        parsed_path = PurePath(path)
+        if path != "/":
+            base.title += " " + str(PurePath(*parsed_path.parts[1:]))
```

Comment:
We prefer f-strings as far as possible. We can create an f-string for the right hand side here.